crates/sparrow-interfaces/src/error.rs

```diff
@@ -0,0 +1,17 @@
+#[derive(derive_more::Display, Debug)]
```

Comment:
I wonder if we want a single top-level errors for this crate, vs. a per-trait error. Eg., I could see us having a `SourceError` for errors coming out of sources. It's kind of nice because it lets us say "sources should return this set of errors, which allows anything that calls the source to actually understand on that", rather than having each source define a separate error enum that is opaque to the user.

Reply:
Yeah, I'm also not against this idea though. The downside being that if we support many sources, that Error file will get large and perhaps confusing knowing when to use what error. But we can figure that out easily if we want -- prefix parquet sources errors with `Error::Parquet..`, `Error::Pulsar..`, if we want. 

I'm ambivalent here. To keep `sparrow-interfaces` simpler with more general errors, or to simplify errors by having them all in one place :shrug:

Reply:
I'd lean towards the source errors being more general so it shouldn't matter (as much) what sources we support. Eg., it's almost like HTTP codes -- `NotFound` could be used by ParquetSource to say "s3 bucket doesn't exist" or PulsarSource to say "topic doesn't exist". This then let's us surface to the user something meaningful (failed because the source couldn't be found).